map_view/show-location-history/src/main/java/com/esri/samples/show_location_history/ShowLocationHistorySample.java

```diff
@@ -0,0 +1,201 @@
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Show Location History Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a map with a dark gray canvas basemap
+      ArcGISMap map = new ArcGISMap("https://www.arcgis.com/home/item.html?id=1970c1995b8f44749f4b9b6e81b5ba45");
+
+      // create a map view and set its map
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // set the map views's viewpoint centered on Los Angeles, California and scaled
+      Point center = new Point(-13185535.98, 4037766.28, SpatialReference.create(3857));
+      mapView.setViewpoint(new Viewpoint(center, 7000));
+
+      // create a button that toggles the location tracking
+      Button trackingButton = new Button("Start tracking");
+      trackingButton.setDisable(true);
+
+      // enable the button interactions when the map is loaded
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
+          trackingButton.setDisable(false);
+        } else {
+          new Alert(Alert.AlertType.ERROR, "Map failed to load").show();
+        }
+      });
+
+      // create a graphics overlay for the points and use a red circle for the symbols
+      GraphicsOverlay locationHistoryOverlay = new GraphicsOverlay();
+      SimpleMarkerSymbol locationSymbol = new SimpleMarkerSymbol(
+        SimpleMarkerSymbol.Style.CIRCLE, 0xFFFF0000, 10f);
+      SimpleRenderer locationHistoryRenderer = new SimpleRenderer(locationSymbol);
+      locationHistoryOverlay.setRenderer(locationHistoryRenderer);
+
+      // create a graphics overlay for the lines connecting the points and use a green line for the symbol
+      GraphicsOverlay locationHistoryLineOverlay = new GraphicsOverlay();
+      SimpleLineSymbol locationLineSymbol = new SimpleLineSymbol(
+        SimpleLineSymbol.Style.SOLID, 0xFF00FF00, 2.0f);
+      SimpleRenderer locationHistoryLineRenderer = new SimpleRenderer(locationLineSymbol);
+      locationHistoryLineOverlay.setRenderer(locationHistoryLineRenderer);
+
+      // add the graphics overlays to the map view
+      mapView.getGraphicsOverlays().addAll(Arrays.asList(locationHistoryOverlay, locationHistoryLineOverlay));
+
+      // create a polyline builder to connect the location points
+      PolylineBuilder polylineBuilder = new PolylineBuilder(SpatialReference.create(3857));
+      // access the json of the location points
+      String polylineData = IOUtils.toString(getClass().getResourceAsStream("/show_location_history/polyline_data.json"),
+        StandardCharsets.UTF_8);
+      // create a polyline from the location points
+      Polyline routePolyline = (Polyline) Polyline.fromJson(polylineData,
```

Comment:
This is a bit of a tricky statement. If you're creating a `Polyline.fromJson`, then there's technically no need to cast it to a `Polyline`, since it's an object of that class already.

But, we can't do ` Polyline routePolyline = Polyline.fromJson(),`, because the return type of that is a 'Geometry'. This is because that method is indirectly referring to `Geometry.fromJson`, which is the type from which `Polyline` inherits. 

So to keep with the class structure, it's best to do the following:

```suggestion
      Polyline routePolyline = (Polyline) Geometry.fromJson(polylineData,
```

That way we're accessing the method `fromJson` at it's lowest level, and trying to cast the result to a Polygon. If there's an error somewhere there, the message might also make it a little easier to understand why something broke (either with creating a generic `Geometry` from the Json, or with casting it to a Polyline)